### PR TITLE
Feature question model

### PR DESCRIPTION
**Pull request by mattfreshwaters**
Just added question model, let me know if it looks fine - merging into Staging

### CODE REVIEW

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    private LocalDate date;     // Store the date of when the question was asked
+
+    public Question(){
+
+    }
+
+    public Question(Question that){
+        this.questionId = that.questionId;
+        this.itemId = that.questionId;
```
Comment (timluu) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:30]: should be *.itemId

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
+package com.talentpath.shamazin.showItemPage.Models;
```
Comment (timluu) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:1]: any chance we can use your pull request to change the folder on this from "Models" to "models" to match the naming convention of the other folders?

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    private Integer votes;      // To store the number of up/downvotes a question receives
+
+    private LocalDate date;     // Store the date of when the question was asked
+
+    public Question(){
+
+    }
+
+    public Question(Question that){
```
Comment (timluu) [inline showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:28]: this is just my general nitpick, so it's totally fine to just leave it as is tbh, but i think renaming "that" to something like "question" would make the object we're referring to a lot more clear and overall readable. although, i suppose "question.question" looks a bit silly, so it's up to you. i think we should just make sure everyone else is on the same page if this is the naming convention we decide to go with

**Review by timluu — APPROVED** [review]
minor nitpicks, but overall, LGTM 👍🏻

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java (hunk 2 above)
Comment (mattfreshwaters) [reply showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:1]: Sure, do you want me to cancel the pull refactor that package then resubmit the pull req?

Comment (timluu) [reply showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:1]: i think you might be able to just refactor on your local repo and push the change to this pull req

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java (hunk 3 above)
Comment (mattfreshwaters) [reply showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:28]: I was just looking up some old code form training & found that we used 'that' in constructors in some old cold for hangman, etc. so figured I'd go with that but i can always change it

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java (hunk 1 above)
Comment (mattfreshwaters) [reply showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:30]: Ah good catch

File: showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java (hunk 3 above)
Comment (timluu) [reply showItemPageBE/src/main/java/com/talentpath/shamazin/showItemPage/Models/Question.java:28]: ahh, got it. works for me then!

**Review by FeiYGH — APPROVED** [review]
looks fine

**Pull request merged**